libnetwork/portallocator/portallocator.go

```diff
@@ -199,7 +200,7 @@ func (p *PortAllocator) ReleasePort(ip net.IP, proto string, port int) {
 	defer p.mutex.Unlock()
 
 	if ip == nil {
-		ip = defaultIP
+		ip = p.defaultIP // FIXME(thaJeztah): consider making this a required argument and producing an error instead, or set default when constructing.
```

Comment:
For this one, there's a bit more code involved but I think we're all good too (ie. we can remove this defaulting).

`ReleasePorts` is called from two places:

1. `attemptBindHostPorts`, in a defer while iterating on the `addrs` slice, which is built from `portBindingReq.childHostIP`. This field is set in `setChildHostIP`. This method either re-use PBs `HostIP` as-is (they already get a default value applied), or uses what the `portDriverClient.ChildHostIP()` returns (which is a `netip.Addr`).

https://github.com/moby/moby/blob/609c6987f1415e2970bd2892429b5309903ffdd4/libnetwork/drivers/bridge/port_mapping_linux.go#L373-L386

https://github.com/moby/moby/blob/609c6987f1415e2970bd2892429b5309903ffdd4/libnetwork/drivers/bridge/port_mapping_linux.go#L400-L421

https://github.com/moby/moby/blob/609c6987f1415e2970bd2892429b5309903ffdd4/libnetwork/drivers/bridge/port_mapping_linux.go#L424-L432

2. `releasePortBindings`, which is called:
    a. in a defer in `addPortMappings` with the result of `bindHostPorts() -> attemptBindHostPorts()`.
    b. from `releasePorts` with the list of port mappings that were effectively created.

Maybe it's worth changing the signature of `ReleasePorts` to make sure no nil `net.IP` get passed.

I think in the future we should also consider replacing `net.IP` with `netip.Addr`.

Reply:
Yup; I'd be +1 for sure to make it a hard failure. I was on the fence changing it already, but went looking where the value came from, and there were quite some places where it _could_ be set, so I decided to (for now) leave a TODO here and to leave it for a follow-up.

(Also because I didn't want to shove such a change in a general "cleanup").